sumo_rl/environment/env.py

```diff
@@ -207,7 +246,88 @@ def _discretize_elapsed_time(self, elapsed):
         for i in range(self.max_green//self.delta_time):
             if elapsed <= self.delta_time + i*self.delta_time:
                 return i
-        return self.max_green//self.delta_time -1
+        return self.max_green//self.delta_time - 1
 
 
-    
+class SumoEnvironmentPZ(AECEnv, EzPickle):
+    metadata = {'render.modes': [], 'name': "sumo_rl_v0"}
+
+    def __init__(self, **kwargs):
+        EzPickle.__init__(self, **kwargs)
+        self._kwargs = kwargs
+
+        self.seed()
+
+        self.agents = self.env.ts_ids
+        self.possible_agents = self.env.ts_ids
+        self._agent_selector = agent_selector(self.agents)
+        self.agent_selection = self._agent_selector.reset()
+        # spaces
+        self.action_spaces = {a: self.env.action_spaces(a) for a in self.agents}
+        self.observation_spaces = {a: self.env.observation_spaces(a) for a in self.agents}
+
+        # dicts
+        self.rewards = {a: 0 for a in self.agents}
+        self.dones = {a: False for a in self.agents}
+        self.infos = {a: {} for a in self.agents}
+
+    def seed(self, seed=None):
+        #self.randomizer, seed = seeding.np_random(seed)
+        self.env = SumoEnvironment(**self._kwargs)
+
+    def reset(self):
+        self.env.reset()
+        self.agents = self.possible_agents[:]
+        self.agent_selection = self._agent_selector.reset()
+        self.rewards = {agent: 0 for agent in self.agents}
+        self._cumulative_rewards = {agent: 0 for agent in self.agents}
+        self.dones = {agent: False for agent in self.agents}
+        self.infos = {agent: {} for agent in self.agents}
+        return self.env.observations.copy()
+
+    def observe(self, agent):
+        obs = self.env.observations[agent].copy()
+        return obs
+
+    def state(self):
+        raise NotImplementedError('Method state() currently not implemented.')
+
+    def close(self):
+        self.env.close()
+
+    def render(self, mode='human'):
+        return self.env.render(mode)
+
+    def step(self, action):
+        if self.dones[self.agent_selection]:
+            return self._was_done_step(action)
+        agent = self.agent_selection
+        if not self.action_spaces[agent].contains(action):
+            raise Exception('Action for agent {} must be in Discrete({}).'
+                            'It is currently {}'.format(agent, self.action_spaces[agent].n, action))
+
+        self.env._apply_actions({agent: action})
+
+        if self._agent_selector.is_last():
+            self.env._run_steps()
+            self.env._compute_observations()
+            new_rewards = self.env._compute_rewards()
+            self.rewards.update(new_rewards)
+        else:
+            self._clear_rewards()
+        
+        self.dones = self.env._compute_dones()
+        if self.dones['__all__']:
+            self.agents = ()
```

Comment:
I feel like this logic shouldn't work. The agents should be removed one at a time with the `_was_done_step` helper method.

Reply:
I.e., there should be no special code that is needed to remove agents.

Reply:
I think the correct logic here is 
```
if self.dones['__all__']:
    del self.dones['__all__']
    for a in self.dones:
         self.dones[a] = True
```

Reply:
Thanks! Fixed it in https://github.com/LucasAlegre/sumo-rl/pull/45/commits/9f91bec2c8806722252c864d96e43c83fe266c4a